File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/inventory/unit_type_conversion.dart
import 'package:farmforge_client/models/inventory/unit_type.dart';

class UnitTypeConversion {
  int unitTypeConversionId;
  int fromUnitId;
  UnitType fromUnit;
  int toUnitId;
  UnitType toUnit;
  int fromQuantity;
  int toQuantity;

  UnitTypeConversion({
    this.unitTypeConversionId = 0,
    this.fromUnitId,
    this.fromUnit,
    this.toUnitId,
    this.toUnit,
    this.fromQuantity,
    this.toQuantity
  });

  factory UnitTypeConversion.fromMap(Map<String, dynamic> data) {
    UnitType fromUnit = data['FromUnit'] != null
      ? UnitType.fromMap(data['FromUnit'])
      : null;

    UnitType toUnit = data['ToUnit'] != null
      ? UnitType.fromMap(data['ToUnit'])
      : null;

    return UnitTypeConversion(
      unitTypeConversionId: data['UnitTypeConversionId'],
      fromUnitId: data['FromUnitId'],
      fromUnit: fromUnit,
      toUnitId: data['ToUnitId'],
      toUnit: toUnit,
      fromQuantity: data['FromQuantity'],
      toQuantity: data['ToQuantity']
    );
  }

  Map<String, dynamic> toMap() {
    Map<String, dynamic> fromUnit = this.fromUnit?.toMap();
    Map<String, dynamic> toUnit = this.toUnit?.toMap();

    return {
      'UnitTypeConversionId': this.unitTypeConversionId,
      'FromUnitId': this.fromUnitId,
      'FromUnit': fromUnit,
      'ToUnitId': this.toUnitId,
      'ToUnit': toUnit,
      'FromQuantity': this.fromQuantity,
      'ToQuantity': this.toQuantity
    };
  }
}

File: StykerDG.FarmForge.DataModel/Models/UnitTypeConversion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("UnitTypeConversion")]
    public class UnitTypeConversion : BaseModel
    {
        [Key]
        public int UnitTypeConversionId { get; set; }
        public int FromUnitId { get; set; }
        [ForeignKey("FromUnitId")]
        public UnitType FromUnit { get; set; }
        public int ToUnitId { get; set; }
        [ForeignKey("ToUnitId")]
        public UnitType ToUnit { get; set; }
        public int FromQuantity { get; set; }
        public int ToQuantity { get; set; }
    }
}
